default.nix

```diff
@@ -52,19 +52,45 @@ let
     };
   };
   jse = pkgs.haskell.lib.justStaticExecutables;
-in with pkgs; rec {
- stack2nix = pkgs.stack2nix;
- hies = runCommandNoCC "hies" {
-   buildInputs = [ makeWrapper ];
- } ''
-   mkdir -p $out/bin
-   ln -s ${hie82}/bin/hie $out/bin/hie-8.2
-   ln -s ${hie84}/bin/hie $out/bin/hie-8.4
-   ln -s ${hie86}/bin/hie $out/bin/hie-8.6
-   makeWrapper ${hie84}/bin/hie-wrapper $out/bin/hie-wrapper \
-     --prefix PATH : $out/bin
- '';
- hie82 = jse (import ./ghc-8.2.nix { inherit pkgs; }).haskell-ide-engine;
- hie84 = jse hie84Pkgs.haskell-ide-engine;
- hie86 = jse hie86Pkgs.haskell-ide-engine;
+
+  wrapWithGHC = ghc: hie: xtra:
+    pkgs.runCommandNoCC "hie${ghc.version}-wrapped" { buildInputs = [ pkgs.makeWrapper ]; } ''
+      mkdir -p $out/bin
+      makeWrapper ${hie}/bin/hie $out/bin/hie \
+        --prefix PATH : ${ghc}/bin
+
+      ${xtra}
+    '';
+
+in
+
+rec {
+
+  stack2nix = pkgs.stack2nix;
+
+  hies = pkgs.runCommandNoCC "hies" {
+    buildInputs = [ pkgs.makeWrapper ];
+  } ''
+    mkdir -p $out/bin
+
+    ln -s ${hie82}/bin/hie $out/bin/hie-8.2
+    ln -s ${hie84}/bin/hie $out/bin/hie-8.4
+    ln -s ${hie86}/bin/hie $out/bin/hie-8.6
+
+    makeWrapper ${hie84}/bin/hie-wrapper $out/bin/hie-wrapper \
+      --prefix PATH : $out/bin \
+      --prefix PATH : ${pkgs.haskell.compiler.ghc844}/bin
```

Comment:
shouldn't this wrap all ghc executables, not just 8.4.4?

Reply:
No since hie-wrapper is only from the 84 one.

Reply:
My understanding is that hie-wrapper will determine the version of GHC being used and then use appropriate GHC/HIE verisons.

Reply:
`hie-wrapper` will execute a `hie-${ghc-version}` executable from the PATH. So as long as each `hie-${ghc-version}` is wrapped with the respective GHC on its PATH then `hie-wrapper` should work accordingly.

I've also used hie-wrapper with a GHC-8.2.x project so I know it works.

Reply:
In that case wrapping `hie-wrapper` with 844 is unnecessary due to hie84 already being wrapped?

Reply:
@domenkozar yeah, I think you're right. I'll drop the wrapper and test it later this evening.

Reply:
Cool, will merge once that's done :) Thanks!

Reply:
Hmm, when I remove the GHC wrapper it complains it cannot find GHC; I haven't looked to see why it depends on it so when I have some free-time I'll look at why but I think it's necessary.